### PR TITLE
Port servicing fix

### PR DESCRIPTION
**Pull request by tarekgh**
This is porting the servicing fix https://github.com/dotnet/coreclr/commit/8ce622d569fae9feb160399db03dc2b86577be1b with a small delta change to get rid of unsafe code in CultureData per Jan recommendation

### CODE REVIEW

**Comment by tarekgh** [comment]
@jkotas @krwq could you please have a look?

**Comment by tarekgh** [comment]
test Ubuntu x64 Checked corefx_baseline please
test Windows_NT x64 Checked corefx_baseline please

File: src/mscorlib/src/System/Globalization/CultureData.cs [hunk 1]
```diff
@@ -626,14 +626,19 @@ internal static CultureData GetCultureData(String cultureName, bool useUserOverr
...
             int i = 0;
 
-            Debug.Assert(name.Length <= LOCALE_NAME_MAX_LENGTH);
+            if (name.Length > LOCALE_NAME_MAX_LENGTH)
+            {
+                // Theoretically we shouldn't hit this exception.
+                throw new ArgumentException(SR.Format(SR.Argument_InvalidId, nameof(name)));
+            }
+
+            Span<char> pName = stackalloc char[name.Length];
```
Comment (jkotas) [inline src/mscorlib/src/System/Globalization/CultureData.cs]: It is not a pointer anymore. Could you please rename this to `neutralName` ?

Comment (tarekgh) [reply src/mscorlib/src/System/Globalization/CultureData.cs]: will do.

**Comment by tarekgh** [comment]
test Ubuntu x64 Checked corefx_baseline please
test Windows_NT x64 Checked corefx_baseline please

**Pull request merged**